### PR TITLE
Configurable header and footer

### PR DESCRIPTION
**Pull request by folix-01**

### CODE REVIEW

File: src/collective/volto/formsupport/browser/send_mail_template.pt [hunk 1]
```diff
@@ -11,7 +16,12 @@
       </li>
     </tal:field>
   </ul>
-  <p i18n:translate="send_mail_text">
-    A new form has been submitted from <a href="${url}"><strong i18n:name="url">${url}</strong></a>
-  </p>
+  <tal:element tal:condition="python:not block.get('mail_footer', '')">
```
Comment (cekk) [inline src/collective/volto/formsupport/browser/send_mail_template.pt:19]: why does this is not inside a <footer> tag if is the default when it's not set?

File: src/collective/volto/formsupport/browser/send_mail_template_table.pt [hunk 2]
```diff
@@ -2,14 +2,21 @@
...
     th {
         text-align: start;
     }
   </style>
+  <header>
+    <tal:element tal:replace="structure python:block.get('mail_header', '')"></tal:element>
+  </header>
   <table border="1">
-    <caption i18n:translate="send_mail_text_table">Form submission data for ${title}</caption>
+    <tal:element tal:condition="python:not block.get('mail_header', '')">
```
Comment (cekk) [inline src/collective/volto/formsupport/browser/send_mail_template_table.pt:17]: Is this a default when mail_header is not set? if yes, it should be inside the same <header> tag and not here.

File: src/collective/volto/formsupport/restapi/services/submit_form/post.py [hunk 3]
```diff
@@ -413,6 +413,7 @@ def prepare_message(self):
             "parameters": self.filter_parameters(),
             "url": self.context.absolute_url(),
             "title": self.context.Title(),
+            "block": self.block,
```
Comment (cekk) [inline src/collective/volto/formsupport/restapi/services/submit_form/post.py:416]: do not pass the entire block, but only the pieces that you need.

**Review by cekk — CHANGES_REQUESTED** [review]
Do not merge until its tested with the frontend.

File: src/collective/volto/formsupport/restapi/services/submit_form/post.py (hunk 3 above)
Comment (folix-01) [reply src/collective/volto/formsupport/restapi/services/submit_form/post.py:416]: no

File: src/collective/volto/formsupport/browser/send_mail_template_table.pt (hunk 2 above)
Comment (folix-01) [reply src/collective/volto/formsupport/browser/send_mail_template_table.pt:17]: what do u mean, example?

File: src/collective/volto/formsupport/restapi/services/submit_form/post.py (hunk 3 above)
Comment (folix-01) [reply src/collective/volto/formsupport/restapi/services/submit_form/post.py:416]: u also have view and portal in the view if i remember, if u pass the block is not changing the performance and keeps the logics so as the view is about the compiled block.

File: src/collective/volto/formsupport/browser/send_mail_template.pt (hunk 1 above)
Comment (folix-01) [reply src/collective/volto/formsupport/browser/send_mail_template.pt:19]: what do u mean, example?

**Comment by cekk** [comment]
LGTM

**Review by cekk — APPROVED** [review]

**Comment by mamico** [comment]
rif: https://github.com/collective/volto-form-block/pull/105

File: src/collective/volto/formsupport/browser/send_mail_template_table.pt [hunk 4]
```diff
@@ -1,18 +1,24 @@
...
+            mail_footer python:options.get('mail_footer', '');
+            ">
   <style>
     th {
         text-align: start;
     }
   </style>
+  <header>
+    <tal:element tal:replace="structure mail_header"></tal:element>
+    <tal:element tal:condition="python:not mail_footer">
```
Comment (mamico) [inline src/collective/volto/formsupport/browser/send_mail_template_table.pt:17]: @folix-01 is it correct `mail_footer` or bettere `mail_header` here?

```suggestion
    <tal:element tal:condition="python:not mail_header">
```

File: src/collective/volto/formsupport/restapi/services/submit_form/post.py [hunk 5]
```diff
@@ -418,6 +419,14 @@ def send_data(self):
                 self.send_mail(msg=acknowledgement_mail, charset=charset)
 
     def prepare_message(self):
+
+        mail_header = self.block.get("mail_header", {}).get("data", "")
+        mail_footer = self.block.get("mail_footer", {}).get("data", "")
+
+        # Check if there is content
+        mail_header = BeautifulSoup(mail_header).get_text() and mail_header or None
+        mail_footer = BeautifulSoup(mail_footer).get_text() and mail_footer or None
```
Comment (mamico) [inline src/collective/volto/formsupport/restapi/services/submit_form/post.py:428]: mail_header = BeautifulSoup(mail_header).get_text() if mail_header else None
        mail_footer = BeautifulSoup(mail_footer).get_text() if mail_footer else None

Comment (mamico) [inline src/collective/volto/formsupport/restapi/services/submit_form/post.py:428]:
```suggestion
        mail_header = BeautifulSoup(mail_header).get_text() if mail_header else None
        mail_footer = BeautifulSoup(mail_footer).get_text() if mail_footer else None
```

**Pull request merged**